File: MDVision.IEHR/Controls/Admin/Admin_EDISubmitInsurance_Detail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MDVision.Datasets;
using MDVision.Business.BCommon;
using System.Data;

using Newtonsoft.Json;
using MDVision.Common.Utilities;
using MDVision.Common.Shared;
using MDVision.Business.BLL;
using MDVision.IEHR.Common;

namespace MDVision.IEHR.Controls.Admin
{
    public class Admin_EDISubmitInsurance_Detail
    {
        private BLLAdminEDI BLLAdminEDIObj = null;
        public Admin_EDISubmitInsurance_Detail()
        {
            BLLAdminEDIObj = new BLLAdminEDI();
        }
        #region Singleton
        private static Admin_EDISubmitInsurance_Detail _obj = null;
        public static Admin_EDISubmitInsurance_Detail Instance()
        {
            if (_obj == null)
                _obj = new Admin_EDISubmitInsurance_Detail();
            return _obj;
        }
        #endregion

        #region Private Functions
        /// <summary>
        /// Saves the edi submit insurance.
        /// </summary>
        /// <param name="fieldsJSON">The fields json.</param>
        /// <returns></returns>
        private string SaveEDISubmitInsurance(string fieldsJSON)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("EDI Submit Insurance", "ADD")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    System.Web.Script.Serialization.JavaScriptSerializer ser = new System.Web.Script.Serialization.JavaScriptSerializer();
                    var SearchedfieldsJSON = ser.Deserialize<dynamic>(fieldsJSON);

                    DSEDI dsEDI = new DSEDI();
                    DSEDI.EDISubmitInsuranceRow dr = dsEDI.EDISubmitInsurance.NewEDISubmitInsuranceRow();

                    dr.SubmitInsuranceName = SearchedfieldsJSON["txtSubmitInsuranceName"];
                    if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlClearinghouse"]))
                        dr.ClearingHouseId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlClearinghouse"]);
                    dr.PayorId = SearchedfieldsJSON["txtPayorId"];
                    dr.PhoneNo = SearchedfieldsJSON["txtTelephone"];
                    dr.PhoneExt = SearchedfieldsJSON["txtExt"];
                    dr.AdmissionDateRequired = MDVUtility.ToStr(SearchedfieldsJSON["chkAdmissionDateRequired"]) == "True" ? true : false;
                    dr.AnesthesiaByMins = MDVUtility.ToStr(SearchedfieldsJSON["chkAnesthesiaByMinutes"]) == "True" ? true : false;
                    dr.IsActive = MDVUtility.ToStr(SearchedfieldsJSON["chkActive"]) == "True" ? true : false;
                    dr.CreatedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                    dr.CreatedOn = DateTime.Now;
                    dr.ModifiedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                    dr.ModifiedOn = DateTime.Now;

                    #region Database Insertion
                    dsEDI.EDISubmitInsurance.AddEDISubmitInsuranceRow(dr);
                    BLObject<DSEDI> obj = BLLAdminEDIObj.InsertEDISubmitInsurance(dsEDI);
                    dsEDI = obj.Data;
                    if (obj.Data != null)
                    {
                        var response = new
                        {
                            status = true,
                            message = Common.AppPrivileges.Save_Message,
                            EDISubmitInsuranceId = dsEDI.Tables[dsEDI.EDISubmitInsurance.TableName].Rows[0][dsEDI.EDISubmitInsurance.EDISubmitIDColumn.ColumnName]
                        };
                        return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = obj.Message
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    #endregion
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Updates the edi submit insurance.
        /// </summary>
        /// <param name="fieldsJSON">The fields json.</param>
        /// <param name="EDISubmitInsuranceId">The edi submit insurance identifier.</param>
        /// <returns></returns>
        private string UpdateEDISubmitInsurance(string fieldsJSON, Int64 EDISubmitInsuranceId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("EDI Submit Insurance", "EDIT")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    System.Web.Script.Serialization.JavaScriptSerializer ser = new System.Web.Script.Serialization.JavaScriptSerializer();
                    var SearchedfieldsJSON = ser.Deserialize<dynamic>(fieldsJSON);

                    DSEDI dsEDI = new DSEDI();
                    //DSEDI.EDISubmitInsuranceRow dr = dsEDI.EDISubmitInsurance.NewEDISubmitInsuranceRow();
                    BLObject<DSEDI> objLoad = BLLAdminEDIObj.LoadEDISubmitInsurance(EDISubmitInsuranceId, null, null, null, null);
                    dsEDI = objLoad.Data;
                    foreach (DSEDI.EDISubmitInsuranceRow dr in dsEDI.Tables[dsEDI.EDISubmitInsurance.TableName].Rows)
                    {
                        //dr.EDISubmitID = EDISubmitInsuranceId;
                        dr.SubmitInsuranceName = SearchedfieldsJSON["txtSubmitInsuranceName"];
                        if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlClearinghouse"]))
                            dr.ClearingHouseId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlClearinghouse"]);
                        dr.PayorId = SearchedfieldsJSON["txtPayorId"];
                        dr.PhoneNo = SearchedfieldsJSON["txtTelephone"];
                        dr.PhoneExt = SearchedfieldsJSON["txtExt"];
                        dr.AdmissionDateRequired = MDVUtility.ToStr(SearchedfieldsJSON["chkAdmissionDateRequired"]) == "True" ? true : false;
                        dr.AnesthesiaByMins = MDVUtility.ToStr(SearchedfieldsJSON["chkAnesthesiaByMinutes"]) == "True" ? true : false;
                        dr.IsActive = MDVUtility.ToStr(SearchedfieldsJSON["chkActive"]) == "True" ? true : false;
                        //dr.CreatedBy = "";
                        //dr.CreatedOn = DateTime.Now;
                        dr.ModifiedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                        dr.ModifiedOn = DateTime.Now;
                    }

                    #region Database Updation
                    //dsEDI.EDISubmitInsurance.AddEDISubmitInsuranceRow(dr);
                    //dsEDI.EDISubmitInsurance.AcceptChanges();

                    if (dsEDI.Tables[dsEDI.EDISubmitInsurance.TableName].Rows.Count > 0)
                    {
                        //dsEDI.EDISubmitInsurance.Rows[0].SetModified();
                        BLObject<DSEDI> obj = BLLAdminEDIObj.UpdateEDISubmitInsurance(dsEDI);
                        if (obj.Data != null)
                        {
                            var response = new
                            {
                                status = true,
                                message = Common.AppPrivileges.Update_Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Message
                            };
                            return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                        }
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = ""
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    #endregion
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Fills the edi submit insurance.
        /// </summary>
        /// <param name="EDISubmitInsuranceId">The edi submit insurance identifier.</param>
        /// <returns></returns>
        private string FillEDISubmitInsurance(Int64 EDISubmitInsuranceId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("EDI Submit Insurance", "VIEW")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    if (string.IsNullOrEmpty(MDVUtility.ToStr(EDISubmitInsuranceId)))
                    {
                        var response = new
                        {
                            status = false,
                            Message = MDVUtility.ToStr(Common.AppPrivileges.Select_Message)
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    else
                    {
                        DSEDI dsEDI = null;
                        BLObject<DSEDI> obj = BLLAdminEDIObj.LoadEDISubmitInsurance(EDISubmitInsuranceId, null, null, null, null);
                        if (obj.Data != null)
                        {
                            dsEDI = obj.Data;
                            if (dsEDI.Tables[dsEDI.EDISubmitInsurance.TableName].Rows.Count > 0)
                            {
                                DataRow dr = dsEDI.Tables[dsEDI.EDISubmitInsurance.TableName].Rows[0];
                                var keyValues = new Dictionary<string, string>
                        {
                            { "ddlClearinghouse", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.ClearingHouseIdColumn.ColumnName])},
                            { "txtSubmitInsuranceName", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.SubmitInsuranceNameColumn.ColumnName])},
                            { "txtPayorId", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.PayorIdColumn.ColumnName])},
                            { "txtTelephone", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.PhoneNoColumn.ColumnName])},
                            { "txtExt", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.PhoneExtColumn.ColumnName])},
                            { "chkAdmissionDateRequired", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.AdmissionDateRequiredColumn.ColumnName])},
                            { "chkAnesthesiaByMinutes", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.AnesthesiaByMinsColumn.ColumnName])},
                            { "chkActive", MDVUtility.ToStr(dr[dsEDI.EDISubmitInsurance.IsActiveColumn.ColumnName])}
                        };
                                System.Web.Script.Serialization.JavaScriptSerializer js = new System.Web.Script.Serialization.JavaScriptSerializer();
                                var response = new
                                {
                                    status = true,
                                    EDISubmitInsuranceFill_JSON = js.Serialize(keyValues)
                                };
                                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                            }
                            else
                            {
                                var response = new
                                {
                                    status = false,
                                    Message = Common.AppPrivileges.No_Record_Message,
                                };
                                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                            }
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }

                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Deletes the edi submit insurance.
        /// </summary>
        /// <param name="EDISubmitInsuranceId">The edi submit insurance identifier.</param>
        /// <returns></returns>
        private string DeleteEDISubmitInsurance(Int64 EDISubmitInsuranceId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("EDI Submit Insurance", "DELETE")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    if (string.IsNullOrEmpty(MDVUtility.ToStr(EDISubmitInsuranceId)))
                    {
                        var response = new
                        {
                            status = false,
                            Message = MDVUtility.ToStr(Common.AppPrivileges.CheckBox_Message)
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    else
                    {
                        BLObject<string> obj = BLLAdminEDIObj.DeleteEDISubmitInsurance(MDVUtility.ToStr(EDISubmitInsuranceId));
                        if (obj.Data == "")
                        {
                            var response = new
                            {
                                status = true,
                                Message = Common.AppPrivileges.Delete_Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Data
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Updates the edi submit insurance is active.
        /// </summary>
        /// <param name="EDISubmitInsuranceId">The edi submit insurance identifier.</param>
        /// <param name="IsActive">The is active.</param>
        /// <returns></returns>
        private string UpdateEDISubmitInsuranceIsActive(Int64 EDISubmitInsuranceId, Int64 IsActive)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("EDI Submit Insurance", "EDIT")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    DSEDI dsEDI = null;
                    BLObject<DSEDI> obj = BLLAdminEDIObj.LoadEDISubmitInsurance(EDISubmitInsuranceId, null, null, null, null);
                    dsEDI = obj.Data;
                    if (dsEDI.Tables[dsEDI.EDISubmitInsurance.TableName].Rows.Count > 0)
                    {
                        DataRow dr = dsEDI.Tables[dsEDI.EDISubmitInsurance.TableName].Rows[0];
                        dr[dsEDI.EDISubmitInsurance.IsActiveColumn.ColumnName] = IsActive;

                        BLObject<DSEDI> objEDISubmitInsurance = BLLAdminEDIObj.UpdateEDISubmitInsurance(dsEDI);
                        string successMsg;
                        if (objEDISubmitInsurance.Data != null)
                        {
                            if (IsActive == 0)
                                successMsg = Common.AppPrivileges.Inactive_Message;
                            else
                                successMsg = Common.AppPrivileges.Active_Message;
                            var response = new
                            {
                                status = true,
                                message = successMsg
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = objEDISubmitInsurance.Message
                            };
                            return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                        }
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = obj.Message
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }
        #endregion

        #region Service Command Handler
        /// <summary>
        /// Handle the EDI Submit Insurance Detail Commands and call to the respective methods.
        /// </summary>
        /// <param name="context">The context.</param>
        public void CommandHandler(HttpContext context)
        {
            string cammandAction = context.Request.QueryString["cammandAction"].ToUpper();

            switch (cammandAction)
            {
                case "SAVE_EDI_SUBMIT_INSURANCE":
                    {
                        string fieldsJSON = context.Request["EDISubmitInsuranceData"];
                        string strJSONData = SaveEDISubmitInsurance(fieldsJSON);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "FILL_EDI_SUBMIT_INSURANCE":
                    {
                        string EDISubmitInsuranceID = context.Request["EDISubmitInsuranceID"];
                        string strJSONData = FillEDISubmitInsurance(MDVUtility.ToInt64(EDISubmitInsuranceID));

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "DELETE_EDI_SUBMIT_INSURANCE":
                    {
                        string EDISubmitInsuranceId = context.Request["EDISubmitInsuranceID"];
                        string strJSONData = DeleteEDISubmitInsurance(MDVUtility.ToInt64(EDISubmitInsuranceId));

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "UPDATE_EDI_SUBMIT_INSURANCE":
                    {
                        string fieldsJSON = context.Request["EDISubmitInsuranceData"];
                        Int64 EDISubmitInsuranceID = MDVUtility.ToInt64(context.Request["EDISubmitInsuranceID"]);
                        string strJSONData = UpdateEDISubmitInsurance(fieldsJSON, EDISubmitInsuranceID);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "UPDATE_EDI_SUBMIT_INSURANCE_ACTIVE_INACTIVE":
                    {
                        Int64 EDISubmitInsuranceID = MDVUtility.ToInt64(context.Request["EDISubmitInsuranceID"]);
                        Int64 IsActive = MDVUtility.ToInt64(context.Request["IsActive"]);
                        string strJSONData = UpdateEDISubmitInsuranceIsActive(EDISubmitInsuranceID, IsActive);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
            }
        }
        #endregion
    }
}

File: MDVision.IEHR/Controls/Admin/Admin_EDISubmitInsurance_Detail.js
EDISubmitInsuranceDetail = {
    params: [],
    Load: function (params) {
        EDISubmitInsuranceDetail.params = params;
       
        var self = $('#tblEDISubmitInsuranceDetail');
        self.loadDropDowns(true).done(function () {

            EDISubmitInsuranceDetail.LoadEDISubmitInsurance();
        });
    },

    LoadEDISubmitInsurance: function () {
        if (EDISubmitInsuranceDetail.params.mode == "Add") {

            //serialize data
            $('#frmEDISubmitInsuranceDetail').data('serialize', $('#frmEDISubmitInsuranceDetail').serialize());
            EDISubmitInsuranceDetail.ValidateEDISubmitInsurance();
        }
        else if (EDISubmitInsuranceDetail.params.mode == "Edit") {
            EDISubmitInsuranceDetail.FillEDISubmitInsurance(EDISubmitInsuranceDetail.params.EDISubmitInsuranceId).done(function (response) {
                if (response.status != false) {
                    var EDISubmitInsurance_detail = JSON.parse(response.EDISubmitInsuranceFill_JSON);
                    var self = $("#EDISubmitInsuranceDetail");
                    
                    utility.bindMyJSON(true, EDISubmitInsurance_detail, false, self).done(function () {

                        if (EDISubmitInsurance_detail.chkAdmissionDateRequired == 'True')
                            $("#EDISubmitInsuranceDetail #chkAdmissionDateRequired").attr("checked", true);
                        else
                            $("#EDISubmitInsuranceDetail #chkAdmissionDateRequired").attr("checked", false);
                        if (EDISubmitInsurance_detail.chkAnesthesiaByMinutes == 'True')
                            $("#EDISubmitInsuranceDetail #chkAnesthesiaByMinutes").attr("checked", true);
                        else
                            $("#EDISubmitInsuranceDetail #chkAnesthesiaByMinutes").attr("checked", false);
                        if (EDISubmitInsurance_detail.chkActive == 'True')
                            $("#EDISubmitInsuranceDetail #chkActive").attr("checked", true);
                        else
                            $("#EDISubmitInsuranceDetail #chkActive").attr("checked", false);
                        EDISubmitInsuranceDetail.ValidateEDISubmitInsurance();


                        //serialize data
                        $('#frmEDISubmitInsuranceDetail').data('serialize', $('#frmEDISubmitInsuranceDetail').serialize());

                    });
                  
                }
                else {
                    utility.DisplayMessages(response.Message, 3);
                }
            });
        }
    },

    ValidateEDISubmitInsurance: function () {
        $('#frmEDISubmitInsuranceDetail')
           .bootstrapValidator({
               live: 'disabled',
               message: 'This value is not valid',
               feedbackIcons: {
                   valid: 'glyphicon glyphicon-ok',
                   invalid: 'glyphicon glyphicon-remove',
                   validating: 'glyphicon glyphicon-refresh'
               },
               fields: {
                   ClearingHouse: {
                       group: '.col-sm-4',
                       validators: {
                           notEmpty: {
                               message: ''
                           }
                       }
                   },
                   SubmitInsName: {
                       group: '.col-sm-4',
                       validators: {
                           notEmpty: {
                               message: ''
                           }
                       }
                   },
                   PayorId: {
                       group: '.col-sm-4',
                       validators: {
                           notEmpty: {
                               message: ''
                           }
                       }
                   }
               }
           })
        .on('success.form.bv', function (e) {
            e.preventDefault();
            EDISubmitInsuranceDetail.EDISubmitInsuranceSave();
        });
    },

    EDISubmitInsuranceSave: function () {
        var strMessage = "";
        var self = $("#EDISubmitInsuranceDetail");
        var myJSON = self.getMyJSON();
        if (EDISubmitInsuranceDetail.params.mode == "Add") {
            AppPrivileges.GetFormPrivileges("EDI Submit Insurance", "ADD", "FORM_PRIVILEGE", "GET_FORM_PRIVILEGE", function (strMessage) {
                if (strMessage == "") {
                    EDISubmitInsuranceDetail.SaveEDISubmitInsurance(myJSON).done(function (response) {
                        if (response.status != false) {
                            Admin_EDISubmitInsurance.EDISubmitInsuranceSearch(response.EDISubmitInsuranceId);
                            utility.DisplayMessages(response.message, 1);
                            EDISubmitInsuranceDetail.UnLoad();
                            CacheManager.BindCodes('GetEDISubmitInsurance', true);
                        }
                        else {
                            utility.DisplayMessages(response.Message, 3);
                        }
                    });
                }
                else
                    utility.DisplayMessages(strMessage, 2);
            });
        }
        else if (EDISubmitInsuranceDetail.params.mode == "Edit") {
            AppPrivileges.GetFormPrivileges("EDI Submit Insurance", "EDIT", "FORM_PRIVILEGE", "GET_FORM_PRIVILEGE", function (strMessage) {
                if (strMessage == "") {
                    EDISubmitInsuranceDetail.UpdateEDISubmitInsurance(myJSON, EDISubmitInsuranceDetail.params.EDISubmitInsuranceId).done(function (response) {
                        if (response.status != false) {
                            if (EDISubmitInsuranceDetail.params["ParentCtrl"] != 'insurancePlanDetail')
                                Admin_EDISubmitInsurance.EDISubmitInsuranceSearch(EDISubmitInsuranceDetail.params.EDISubmitInsuranceId);

                            utility.DisplayMessages(response.message, 1);
                            EDISubmitInsuranceDetail.UnLoad();
                            CacheManager.BindCodes('GetEDISubmitInsurance', true);
                        }
                        else {
                            utility.DisplayMessages(response.Message, 3);
                        }
                    });
                }
                else
                    utility.DisplayMessages(strMessage, 2);
            });
        }
    },

    SaveEDISubmitInsurance: function (EDISubmitInsuranceData) {
        var data = "EDISubmitInsuranceData=" + EDISubmitInsuranceData;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_EDI_SUBMIT_INSURANCE_DETAIL", "SAVE_EDI_SUBMIT_INSURANCE");
    },

    UpdateEDISubmitInsurance: function (EDISubmitInsuranceData, EDISubmitInsuranceID) {
        var data = "EDISubmitInsuranceData=" + EDISubmitInsuranceData + "&EDISubmitInsuranceID=" + EDISubmitInsuranceID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_EDI_SUBMIT_INSURANCE_DETAIL", "UPDATE_EDI_SUBMIT_INSURANCE");
    },

    FillEDISubmitInsurance: function (EDISubmitInsuranceID) {
        var data = "EDISubmitInsuranceID=" + EDISubmitInsuranceID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_EDI_SUBMIT_INSURANCE_DETAIL", "FILL_EDI_SUBMIT_INSURANCE");
    },

    UpdateEDISubmitInsuranceActiveInactive: function (EDISubmitInsuranceID, IsActive) {
        var data = "EDISubmitInsuranceID=" + EDISubmitInsuranceID + "&IsActive=" + IsActive;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_EDI_SUBMIT_INSURANCE_DETAIL", "UPDATE_EDI_SUBMIT_INSURANCE_ACTIVE_INACTIVE");
    },

    //UnLoad: function () {

    //    utility.UnLoadDialog("frmEDISubmitInsuranceDetail", function () {
    //        UnloadActionPan();
    //    }, function () {
    //        UnloadActionPan();
    //    });

    //},
    UnLoad: function () {
        if (EDISubmitInsuranceDetail.params["FromAdmin"] == "0" || EDISubmitInsuranceDetail.params["RefCtrl"] == "lnkInsuranceEDI") {


            if (EDISubmitInsuranceDetail.params != null && EDISubmitInsuranceDetail.params.ParentCtrl != null && EDISubmitInsuranceDetail.params.PanelID != 'EDISubmitInsuranceDetail') {
                UnloadActionPan(EDISubmitInsuranceDetail.params.ParentCtrl, 'EDISubmitInsuranceDetail', null, EDISubmitInsuranceDetail.params.PanelID);
            }

            else if (EDISubmitInsuranceDetail.params != null && EDISubmitInsuranceDetail.params.ParentCtrl != null) {
                UnloadActionPan(EDISubmitInsuranceDetail.params.ParentCtrl, 'EDISubmitInsuranceDetail');
            }

            else
                UnloadActionPan(null, 'EDISubmitInsuranceDetail');
        }
        else {
            RemoveAdminTab();
        }
    },
    ShowHistory: function () {
        var PanelID = 'EDISubmitInsuranceDetail';
        var ParentCtrl = 'EDISubmitInsuranceDetail';
        var ProfileName = 'EDI Submit Insurance';
        var DBTableName = 'EDISubmitInsurance';
        var ColumnKeyId = EDISubmitInsuranceDetail.params.EDISubmitInsuranceId;

        utility.ShowHistory(ParentCtrl, PanelID, ProfileName, DBTableName, ColumnKeyId);
    },

}